src/coreclr/tools/Common/TypeSystem/IL/Stubs/ComparerIntrinsics.cs

```diff
@@ -104,6 +104,14 @@ private static InstantiatedType GetComparerForType(TypeDesc type, string flavor,
                 return context.SystemModule.GetKnownType("System.Collections.Generic", $"Enum{flavor}`1")
                     .MakeInstantiatedType(type);
             }
+            else if (IsMultipleCanonInstantiation(type))
```

Comment:
I think changing `ImplementsInterfaceOfSelf` to return `Nullable<bool>` (yes/no/unknown) would work better.